src/module/apps/sidebar/actor-directory.ts

```diff
@@ -49,9 +49,13 @@ class ActorDirectoryPF2e extends ActorDirectory<ActorPF2e<null>> {
         // Strip actor level from actors we lack proper observer permission for
         for (const element of htmlQueryAll(html, "li.directory-item.actor")) {
             const actor = game.actors.get(element.dataset.documentId ?? "");
-            if (!actor?.testUserPermission(game.user, "OBSERVER")) {
-                element.querySelector("span.actor-level")?.remove();
-            }
+            const hideLevel = actor?.isOfType("character", "familiar", "vehicle")
+                ? !actor?.testUserPermission(game.user, "LIMITED")
+                : actor?.isOfType("npc")
+                ? actor.visible
```

Comment:
wouldn't this hide npcs that the player has ownership over?

Reply:
Doh, didn't test thoroughly enough

Reply:
Now it hides stuff from GMs!

Reply:
Honestly I'd probably end up doing `(actor?.isOfType("character", "familiar", "vehicle") && actor?.testUserPermission(game.user, "LIMITED")) || actor?.testUserPermission(game.user, "OBSERVER"))` but I don't know if that causes an issue with not-listed npcs, If it does, then you'd do

```ts
const hasPermission = (actor?.isOfType("character", "familiar", "vehicle") && actor.testUserPermission(game.user, "LIMITED")) || actor?.testUserPermission(game.user, "OBSERVER"));
if (hasPermission && actor?.visible) {
    // blah
}
```

accounting for prettier of course.

Reply:
That said I don't know if its worth getting fiddly to see teammate levels. One might argue seeing the level for a player helps makes it faster to find your own PC on a skim.

Reply:
As it is, it works fine for me. `testUserPermission` will always return true for a GM, which will set `hideLevel` to false, so it won't hide the level. It accounts for NPCs since Limited NPCs don't show level anyway... Although now I think about it we don't need that special check for NPCs; we can default to Observer for all other actors... which is the behavior I had 4 commits ago!

Reply:
As its written now, it excludes hazard levels.